queries/usd/highlights.scm

```diff
@@ -0,0 +1,228 @@
+(None) @constant.builtin
+(asset_path) @string.special.asset
+(bool) @boolean
+(comment) @comment @spell
+(custom) @function.builtin
+(float) @float
+(integer) @number
+(namespace_identifier) @namespace
+(orderer) @function.call
+(prim_path) @string.special.prim
```

Comment:
USD has at least 3-4 different "string-like" representations.

**Strings**
`"foo"` - A regular string. Depending on where in the file this is defined, this is actually a raw string, a docstring, or a comment

**Comments**
`# foo` - This is always an inline comment

**Asset Paths**
There are always external file references (e.g. loading another file)
`@./relative/path.usd@`
`@uri://www.resource.com/inner/path?query=here#fragment@`

**Prim Paths**
These are generally-but-not-always intra-file references (referring to another object in the same file or from another file)
`</path/to/foo_prim>`

**Asset Path + Prim Path**
The above two sections can be combined.
`@./relative/path.usd@</path/to/foo_prim>`

Because Asset Path and Prim Path can be combined or be separate, it would help a lot for readability's sake if they can be differentiated. The nvim-treesitter out-of-box literal recommendations are https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#literals but the `@string` literals don't map well to the different USD cases.

Reply:
I don't have the link handy but I remember reading from either tree-sitter or nvim-treesitter's documentation that it's legal to append more text onto highlight groups and the longest highlight group will match. For people who just have  colorschemes that specify `@string.special`, these groups would work for those. But if someone wanted to add a buffer-local highlight group for just Asset Paths (for example) they still can.

If we can use "out of box" nvim-treesitter highlight groups for each of these 3-4 cases then I'm happy to do that though. If you have recommended alternatives I can try those out. Thank you :)

Reply:
I'm just saying that `prim_path` should be `@string.special`, or something markdown-url-like?. It will be aliased to `@string.special.filetype` anyways (Check your `:Inspect`)

Reply:
Strings - `@string`
Comments - `@comment`
Asset/Prim paths - `@string.special`, `@text.uri`, or `@string`

Reply:
@ObserverOfTime would give better recommendations here.

Reply:
prim_path is now `@string.special` and asset_path is `@text.uri`

Reply:
Our previous policy was to allow language specific suffixes when appropriate (not sure if this changes with unification efforts with helix). Just the prefix should follow contribution.md

Reply:
I thought the language suffixes were automatic. e.g. If I call `:Inspect` while looking at a USD file the full highlight group is `@string.special.usd` even if I define it as `@string.special` in this file

Reply:
I think what was meant here were prefixes that are _not_ the language name but only makes sense for a specific language (like `@string.special.prim` here).

Reply:
@clason Checking back on this comment again - "I'm just saying that prim_path should be @string.special, no need to say .prim, or something markdown-url-like, maybe @text.uri ?. It will be aliased to @string.special.filetype, which is @string.special.prim anyways (Check your :Inspect)" 

Yes it looks like there was that misunderstanding. But I think @amaanq recommendation works still.

Whether we use

prim_path - @string.special 
asset_path - @text.uri

or

prim_path - @string.special.prim
asset_path - @srting.special.asset

I'm okay with either approach as long as the highlights are distinguishable

Reply:
To better clarify my words, all hlgroups that you specify in `highlights.scm` file, for example, `@foo.bar`, when using in a file with ft `bax`, will have the hlgroups `@foo.bar.bax` providing the highlight. By default, this fallbacks to `@foo.bar`.

Reply:
The prim/asset nested capture are, with all due respect, useless as users/themes do **not** account for this. Hence why such an extension is best left up to user configuration/extension.

Imo both should be `@string.special` with one of them tacking on `@text.uri` as well if that makes sense 🤷